GameServer/Logic/GameObjects/GameObjects.cs

```diff
@@ -14,23 +14,27 @@ public class GameObject : Target
         public uint NetId { get; private set; }
         protected float xvector, yvector;
 
-        /**
-         * Current target the object running to (can be coordinates or an object)
-         */
+        /// <summary>
+        /// Current target the object running to (can be coordinates or an object)
+        /// </summary>
         public Target Target { get; set; }
 
         public List<Vector2> Waypoints { get; private set; }
         public int CurWaypoint { get; private set; }
-        public TeamId Team { get; private set; }
-        public void SetTeam(TeamId team)
+        private TeamId _team;
+        public TeamId Team
```

Comment:
Yeah undo the changes here like explained above. Properties shouldn't contain logic, and if logic is needed, make the property read only and have a function for setting it.